src/app/messages/page.tsx

```diff
@@ -1,30 +1,54 @@
+import Button from '@/components/common/Button'
+import Link from 'next/link'
+import Image from 'next/image'
+import { css } from '../../../styled-system/css'
+import MessagePreviewList from '@/components/MessagePreviewList'
+import { Suspense } from 'react'
 import { createServerComponentClient } from '@supabase/auth-helpers-nextjs'
-import { redirect } from 'next/navigation'
 import { Database } from '@/types/supabase'
 import { cookies } from 'next/headers'
+import { getLastMessageByChat } from '../domains/messages/messages.service'
+import { getProfiles } from '../domains/profiles/profiles.service'
+import Loading from './loading'
 
-export default async function MessagesPage() {
+async function MessagesPage() {
   const supabase = createServerComponentClient<Database>({ cookies })
+  const {
+    data: { session },
+  } = await supabase.auth.getSession()
 
-  const { data } = await supabase.auth.getSession()
+  const { data: chats } = await supabase
+    .from('chats')
+    .select('chat_id, users')
+    .contains('users', [session?.user.id])
 
-  if (!data.session) {
-    return redirect('/login')
-  }
-
-  const { data: contacts } = await supabase
-    .from('contacts')
-    .select('contact_id, user_id, contact_user_id')
-    .eq('user_id', data.session.user.id)
+  const lastMessages = await getLastMessageByChat({ chats: chats || [] })
+  const contacts = await getProfiles()
 
   return (
     <>
-      <h1>Messages</h1>
-      <ul>
-        {contacts?.map(contact => (
-          <li key={contact.contact_id}>{contact.contact_user_id}</li>
-        ))}
-      </ul>
+      <Suspense fallback={<Loading />}>
```

Comment:
praise: well done here! visually it's not the most beautiful thing, but the logic is there!

suggestion: remember to always include "empty states" in your UIs. In this case, when there are no messages, it could show the sentence "No messages created yet." But even better than that is turning every empty state into an "action moment", i.e., always suggest an action for the user to do. Here you could add a button below the sentence saying "Send your first text". Here's a [related UX article](https://uxplanet.org/how-to-design-better-empty-states-f8787e2e320b)

Reply:
action moment, I like it, gonna add it!